### PR TITLE
fix: ask user for the camera permission

### PR DESCRIPTION
**Pull request by fhenrich33**
Displays a warning to the user if they don't allow permission to the camera for the QR scanner.

### CODE REVIEW

**Comment by fhenrich33** [comment]
[Bug: If the QR Scanner doesn't have access to the camera, there should be an error message and an ability for users to re-request.](https://trello.com/c/CHbAL8Ha/1420-bug-if-the-qr-scanner-doesnt-have-access-to-the-camera-there-should-be-an-error-message-and-an-ability-for-users-to-re-request)

File: front/src/components/QrReader/QrReaderContainer.tsx
Comment (fhenrich33) [inline front/src/components/QrReader/QrReaderContainer.tsx:160]: I could move the copy to a `const` at the top of the file if this is a more palatable code style (I usually do that.)

```ts
const CAMERA_NOT_PERMITED_TEXT = "Camera permission was not allowed. Please unblock your camera in the address bar, reload the page, and then grant permission to your camera.";
// ...
<AlertWithoutAction alertText={CAMERA_NOT_PERMITED_TEXT}  />
```

**Comment by fhenrich33** [comment]
[Bug: If the QR Scanner doesn't have access to the camera, there should be an error message and an ability for users to re-request.](https://trello.com/c/CHbAL8Ha/1420-bug-if-the-qr-scanner-doesnt-have-access-to-the-camera-there-should-be-an-error-message-and-an-ability-for-users-to-re-request)

File: front/src/components/QrReader/QrReaderContainer.tsx
Comment (HaGuesto) [inline front/src/components/QrReader/QrReaderContainer.tsx:46]: I think you are mocking it already :)

File: front/src/components/QrReader/QrReaderContainer.tsx
Comment (HaGuesto) [inline front/src/components/QrReader/QrReaderContainer.tsx:55]: the triggerError function also includes a console.error. So, I would rather like you to use that.

File: front/src/components/QrReader/QrReaderContainer.tsx
Comment (HaGuesto) [inline front/src/components/QrReader/QrReaderContainer.tsx:61]: the triggerError function also includes a console.error. So, I would rather like you to use that.

**Review by HaGuesto — APPROVED** [review]
Congrats to your first PR @fhenrich33 . Nice work with the mocking.

I usually do a code review first where I left a few small comments, followed by a functional review which passed in your case :)

Pls fix the comments I made in the code,. I already approve the PR since the changes are minor and I do not want to block you by an additional review.

**Comment by aerinsol** [comment]
@fhenrich33 once you're done with the small fixes go ahead and merge it to master. You shouldn't have any issues since it's approved.

**Comment by fhenrich33** [comment]
> @fhenrich33 once you're done with the small fixes go ahead and merge it to master. You shouldn't have any issues since it's approved.

@aerinsol It seems that I can't merge it:
> You’re not [authorized](https://docs.github.com/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches) to merge this pull request.

**Comment by aerinsol** [comment]
@fhenrich33 it looks like one of the FE tests failed in Circle CI. Are you able to get it to pass? If it passes and you still can't merge tell me - I'll merge and take a look at the permissions later.

**Comment by fhenrich33** [comment]
@aerinsol Managed to make the flaky frontend test pass, but now a backend test is failing 😞

**Comment by aerinsol** [comment]
Hmm, functional review blocker as tested in the field - iPhone Safari seems to not have this option to unblock in the address bar and requests permissions every time. Do we need to set a cookie of some sort to make it work.

**Comment by aerinsol** [comment]
Unfortunately the best implementation is described here: https://support.scandit.com/hc/en-us/articles/360008443011-Why-does-iOS-keep-asking-for-camera-permissions

**Comment by fhenrich33** [comment]
@aerinsol Waiting for the copy for iOS. I used this as a placeholder:

```ts
const CAMERA_NOT_PERMITED_TEXT_SAFARI_IOS = "Camera access was denied. Please unblock camera access by tapping the AA icon > Website Settings > set Camera permission to \"Allow\".";
```

**Comment by aerinsol** [comment]
@fhenrich33 looks good, I would do:
Camera access was denied. Please allow camera access in the address bar by selecting _AA_ > _Website Settings_ >  _Camera_ > \"Allow\"

**Pull request merged**
